packages/components/src/components/FunctionChart.tsx

```diff
@@ -23,52 +23,60 @@ export const FunctionChart: React.FC<FunctionChartProps> = ({
-  let result1 = runForeign(fn, [chartSettings.start], environment);
-  let result2 = runForeign(fn, [chartSettings.stop], environment);
-  let getValidResult = () => {
-    if (result1.tag === "Ok") {
-      return result1;
-    } else if (result2.tag === "Ok") {
-      return result2;
-    } else {
-      return result1;
-    }
-  };
-  let validResult = getValidResult();
-  let resultType = validResult.tag === "Ok" ? validResult.value.tag : "Error";
+  if (fn.parameters.length > 1) {
+    return (
+      <MessageBox heading="Function Display Not Supported">
+        Only functions with one parameter are displayed.
+      </MessageBox>
+    );
+  } else {
+    let result1 = runForeign(fn, [chartSettings.start], environment);
+    let result2 = runForeign(fn, [chartSettings.stop], environment);
+    let getValidResult = () => {
+      if (result1.tag === "Ok") {
+        return result1;
+      } else if (result2.tag === "Ok") {
+        return result2;
+      } else {
+        return result1;
+      }
+    };
+    let validResult = getValidResult();
+    let resultType = validResult.tag === "Ok" ? validResult.value.tag : "Error";
 
-  let component = () => {
-    switch (resultType) {
-      case "distribution":
-        return (
-          <FunctionChart1Dist
-            fn={fn}
-            chartSettings={chartSettings}
-            environment={environment}
-            height={height}
-          />
-        );
-      case "number":
-        return (
-          <FunctionChart1Number
-            fn={fn}
-            chartSettings={chartSettings}
-            environment={environment}
-            height={height}
-          />
-        );
-      case "Error":
-        return (
-          <ErrorBox heading="Error">The function failed to be run</ErrorBox>
-        );
-      default:
-        return (
-          <ErrorBox heading="No Viewer">
-            There is no function visualization for this type of function
-          </ErrorBox>
-        );
-    }
-  };
+    let component = () => {
+      switch (resultType) {
+        case "distribution":
+          return (
+            <FunctionChart1Dist
+              fn={fn}
+              chartSettings={chartSettings}
+              environment={environment}
+              height={height}
+            />
+          );
+        case "number":
+          return (
+            <FunctionChart1Number
+              fn={fn}
+              chartSettings={chartSettings}
+              environment={environment}
+              height={height}
+            />
+          );
+        case "Error":
+          return (
+            <ErrorBox heading="Error">The function failed to be run</ErrorBox>
+          );
+        default:
+          return (
+            <MessageBox heading="Function Display Not Supported">
+              There is no function visualization for this type of output
+            </MessageBox>
+          );
+      }
+    };
 
-  return component();
+    return component();
```

Comment:
I know this code was here before, but: what's the point of wrapping `switch` in a function?